### PR TITLE
Run nb-clean in formatting workflow

### LINKED ISSUE #2376: Use nb-clean
Following the discussions in the UX workshop, maybe we should start using https://github.com/srstevenson/nb-clean

It can remove outputs from notebooks automatically when committing them.

### PR DESCRIPTION
**Pull request by jl-wynen**
Fixes #2376

I added dtype.ipynb to check if nb-clean works.
nb-convert successfully removed the output and empty cell from it and preserved the `raises-exception` tag. As an added bonus, it also removes the version of the Python interpreter that last ran the notebook. This was annoyingly getting updated all the time.

There does not seem to be a preconfigured pre-commit hook for it. I didn't spend much time trying to make it work. We should decide if it is worth the effort.

### CODE REVIEW

File: .github/workflows/pr_and_main.yml [hunk 1]
```diff
@@ -16,15 +16,17 @@ jobs:
...
+      # Run this before any other commands so that git diff does not pick up formatting changes.
+      - run: |
+          find . -type f -not -path '.git/*' -exec sed -ri "s/[0-9]{4} (Scipp contributors)/$(date +%Y) \1/g" {} +
+          git diff --exit-code
       - run: sudo apt-get install --yes clang-format-10
       - run: find lib -type f -regex '.*\.\(cpp\|h\|tcc\)' -exec clang-format-10 -i {} +
-      - run: pip install cmake_format==0.6.9 flake8 yapf==0.30.0
+      - run: pip install cmake_format==0.6.9 flake8 nb-clean==2.1.0 yapf==0.30.0
       - run: find . -type f -name CMakeLists.txt -or -name '*.cmake' -not -path "./lib/.tbb/*" | xargs cmake-format -i
+      - run: find docs -type f -regex '.*\.ipynb' | xargs nb-clean clean --remove-empty-cells --preserve-cell-metadata
```
Comment (SimonHeybrock) [inline .github/workflows/pr_and_main.yml:27]: Any reason against running this on the top level folder? There is also https://github.com/scipp/scipp/tree/main/tools/manual_testing that contains a notebook, and there may be others, e.g., if we add code documentation alongside code?

File: scipp-developer.yml [hunk 2]
```diff
@@ -43,6 +43,8 @@ dependencies:
   - pre-commit
   - yapf
   - flake8
+  - pip:
+    - nb-clean
```
Comment (SimonHeybrock) [inline scipp-developer.yml:47]: Now CI will install this twice?

File: .github/workflows/pr_and_main.yml (hunk 1 above)
Comment (jl-wynen) [reply .github/workflows/pr_and_main.yml:27]: No. I just didn't think about notebooks in other locations.

File: scipp-developer.yml (hunk 2 above)
Comment (jl-wynen) [reply scipp-developer.yml:47]: Yes. Just like flake8 and yapf. Is there a way around it?

Comment (SimonHeybrock) [reply scipp-developer.yml:47]: I guess the issue is that we use these `yml` files for CI *and* developers? Maybe with @nvaytet's script we can include the non-CI requirements as extras when making envs for devs?

**Comment by jl-wynen** [comment]
The actions completed successfully. But the report seems to be broken. Have you seen this before?

**Comment by SimonHeybrock** [comment]
Hmm, there is a green checkmark on the previous commit?

**Comment by jl-wynen** [comment]
I guess GH waits for the most recent commit to be checked.

File: .github/workflows/pr_and_main.yml [hunk 3]
```diff
@@ -16,6 +16,8 @@ jobs:
 
     steps:
       - uses: actions/checkout@v2
+        with:
+          ref: ${{ github.head_ref }}
```
Comment (SimonHeybrock) [inline .github/workflows/pr_and_main.yml:20]: This is for avoiding the re-run issue?

**Comment by jl-wynen** [comment]
I checked the documentation of auto-commit. There seems to be a way to trigger new actions from it  but I don't know if that is what we want.
The docs recommend to check out `head_ref` in PRs. Let's see if that helps with the report or rerun problem.

File: .github/workflows/pr_and_main.yml (hunk 3 above)
Comment (jl-wynen) [reply .github/workflows/pr_and_main.yml:20]: Maybe

**Review by SimonHeybrock — APPROVED** [review]

**Comment by SimonHeybrock** [comment]
There is a merge conflict

**Pull request merged**